ClientApp/viking-vault/src/display_user_list/AddMoneyModal.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import './AddMoney.css';
+
+interface IAddMoneyModalProps{
+    open: boolean;
+    userName: string;
+    closeModal: any;
+    addMoney: any;
+}
+
+interface IAddMoneyModalState{
+    amount: string;
+}
+
+class AddMoneyModal extends React.Component<IAddMoneyModalProps, IAddMoneyModalState>{ 
+    state = {
+            amount : ""
+        };
+
+    private resetAmountField(){
+        this.setState({
+            amount : ""
+        });
+    }
+    private closeModal = () =>{
+        this.resetAmountField();
+        this.props.closeModal(false);
+    }
+
+    private handleAddMoney = () =>{
+        this.resetAmountField();
+        this.props.addMoney(Number(this.state.amount));
+    }
+
+    private handleAmountChange = (e:any) =>{
```

Comment:
you can write this in a simplified manner, of course it's only suitable for simple things `private handleAmountChange = (e: any) => this.setState({ amount: e.target.value });`